src/components/atoms/popover/Popover.tsx

```diff
@@ -1,29 +1,42 @@
+import { ExternalLink } from "@/components/ExternalLink";
+import { BaseUIArrow } from "@/utils/baseUI";
 import { joinTailwindClasses } from "@/utils/tailwind";
 import { Popover as BasePopover } from "@base-ui-components/react/popover";
+import Link from "next/link";
 import { ReactElement, ReactNode } from "react";
 
-const PopoverArrow = () => (
-  <svg width="20" height="10" viewBox="0 0 20 10" fill="none">
-    <path
-      d="M9.66437 2.60207L4.80758 6.97318C4.07308 7.63423 3.11989 8 2.13172 8H0V10H20V8H18.5349C17.5468 8 16.5936 7.63423 15.8591 6.97318L11.0023 2.60207C10.622 2.2598 10.0447 2.25979 9.66437 2.60207Z"
-      className="fill-surface-light"
-    />
-    <path
-      d="M10.3333 3.34539L5.47654 7.71648C4.55842 8.54279 3.36693 9 2.13172 9H0V8H2.13172C3.11989 8 4.07308 7.63423 4.80758 6.97318L9.66437 2.60207C10.0447 2.25979 10.622 2.2598 11.0023 2.60207L15.8591 6.97318C16.5936 7.63423 17.5468 8 18.5349 8H20V9H18.5349C17.2998 9 16.1083 8.54278 15.1901 7.71648L10.3333 3.34539Z"
-      className="fill-border-light"
-    />
-  </svg>
-);
-
-interface PopoverProps {
-  children: ReactNode;
+export type TPopoverLink = {
+  href: string;
+  text: string;
+  external?: boolean;
+};
+
+// Provide options for structuring the popover using props & Base UI components (heavily preferred) or children for full flexibility
+
+interface PopoverGenericProps {
   onOpenChange?: (open: boolean) => void;
   openOnHover?: boolean;
   popupClasses?: string;
   trigger: ReactElement;
 }
 
-export const Popover = ({ children, onOpenChange, openOnHover = false, popupClasses = "", trigger }: PopoverProps) => {
+interface PopoverElementProps extends PopoverGenericProps {
+  title?: string;
+  description: string;
+  link?: TPopoverLink;
+  children?: never;
+}
+
+interface PopoverChildrenProps extends PopoverGenericProps {
+  children: ReactNode;
+  title?: never;
+  description?: never;
+  link?: never;
+}
+
+type PopoverProps = PopoverElementProps | PopoverChildrenProps;
```

Comment:
Request: we decide between ourselves a naming convention for types and interfaces.

Personally I quite like the prefix approach so that one instantly knows what the nature of the thing is, e.g. `TTypeName`, `IInterfaceName`, or even `TProps` if it is only used in the component.

Reply:
My usage here is purely out of habit, I agree that it'd be best to standardise.

Personally I don't see a ton of value in using a prefix. There are rarely cases where the type/interface distinction is useful for TypeScript in React, and IDEs make it easy to contextualise the type on hover.

That being said, the prefix is useful at a glance (if a bit ugly), and I like that it adds a bit more intentionality to deciding whether to use a type or interface for an object.